core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -75,6 +75,10 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                 fieldName = currentFieldName;
                 values = parseValues(parser);
             } else if (token == XContentParser.Token.START_OBJECT) {
+                if  (fieldName != null) {
+                    throw new ParsingException(parser.getTokenLocation(), "[terms] query does not support more than one field. "
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?